src/trove/string-dict.scrbl

```diff
@@ -225,6 +236,102 @@ check:
 end
 }
 
+@sd-method["merge"
+  #:contract (a-arrow (SD-of "a") (SD-of "a") (SD-of "a"))
+  #:args (list (list "self" #f) (list "other" #f))
+  #:return (SD-of "a")
+]
+
+Returns a new immutable string-dict that has the keys of both
+the original string-dict and the @pyret{other} string-dict. Choose a key's
+value to be the one it has in the @pyret{other} string-dict, if it has one;
+otherwise it has the value from the original string-dict.
+
+@examples{
+check:
+  sd1 = [string-dict: "a", 5, "c", 4]
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd3 = sd1.merge(sd2)
+  sd3 is [string-dict: "a", 10, "b", 6, "c", 4]
+  sd4 = sd2.merge(sd1)
+  sd4 is [string-dict: "a", 5, "b", 6, "c", 4]
+  sd2.merge(sd1).merge(sd1) is sd2.merge(sd1)
+end
+}
+
+@sd-method["keys-list"
+  #:contract (a-arrow (SD-of "a") (L-of S))
+  #:args (list (list "self" #f))
+  #:return (L-of S)
+]
+
+Returns the list of keys in the immutable string-dict, in some order.
+
+@examples{
+check:
+  sd2 = [string-dict: "a", 10, "b", 6]
+  sd2.keys-list() is [list: "a", "b"]
```

Comment:
If the order is unspecified, then this test might fail.  Either add a `.sort()` or make it into a set or something...